File: code/java/src/main/java/de/activegroup/Dosage.java
package de.activegroup;

public sealed interface Dosage {

    String format2();
    record Tablet(int morning, int midday, int evening)
            implements Dosage {
        @Override
        public String format2() {
            return morning + "-" + midday + "-" + evening;
        }

    }
    record Infusion(double speed, int duration)
            implements Dosage {
        @Override
        public String format2() {
            return speed + "ml/min for " + duration + "h";
        }

    }

}

File: code/kotlin/src/Dosage.kt
sealed interface Dosage {
    fun format(): String =
        when(this) {
            is Tablet ->
                "$morning-$midday-$evening"
            is Infusion ->
                speed.toString() + "ml/min for " + duration + "h"
        }
    fun format2(): String

    data class Tablet(val morning: Int, val midday: Int, val evening: Int) : Dosage {
        override fun format2(): String {
            return "$morning-$midday-$evening"
        }
    }

    data class Infusion(val speed: Double, val duration: Int) : Dosage {
        override fun format2(): String {
            return speed.toString() + "ml/min for " + duration + "h"
        }
    }
}
